src/jit/target.h

```diff
@@ -678,6 +678,12 @@ typedef unsigned short regPairNoSmall; // arm: need 12 bits
 
   #define RBM_ARG_REGS            (RBM_ARG_0|RBM_ARG_1)
 
+  // The registers trashed by profiler enter/leave/tailcall hook
+  // See vm\i386\asmhelpers.asm for more details.
+  #define RBM_PROFILER_ENTER_TRASH     RBM_NONE
+  #define RBM_PROFILER_LEAVE_TRASH     RBM_NONE
+  #define RBM_PROFILER_TAILCALL_TRASH  (RBM_ALLINT & ~RBM_ARG_REGS)
```

Comment:
>#define RBM_PROFILER_TAILCALL_TRASH  (RBM_ALLINT & ~RBM_ARG_REGS) [](start = 2, length = 65)

When I looked at codegenlegacy.cpp, it is emitting tail-call profiler hook after the args are setup.  It also makes sense on x86 since tail call hook doesn't trash arg regs.

Whereas on amd64, tail call profiler hook is emitted before the arguments are setup but after all arg side-effects are computed.  It makes sense on amd64 because once an argument is setup, we cannot generate another call because calls use the same fixed out-going arg space. It is also reflected in TAIL_CALL_TRASH that it doesn't trash only return regs.

Since RyuJIT x86 is going to share the same call lowering and codegen logic for GT_PROF_HOOK, we will be emitting tail-call profiler callback before args are setup, which will be different than what legacy x86 did.
Do we know whether it is an ABI contract on x86 that profiler hook needs to be emitted after call args are setup?